### PR TITLE
Feature/621 make dates required

### PR DESCRIPTION
**Pull request by junie-stigdon**
## Pull Request

- Made beginDate and endDate required
- Commented out date-validation.pipe.ts and replaced it with custom decorators:
  - is-iso-format.pipe.ts
  - is-valid-date.pipe.ts
  - is-in-date-range.pipe.ts
- Removed "endDate >= beginDate" validation from the service and replaced it with a custom decorator:
  - is-date-greater.pipe.ts
- If user enters invalid format, invalid date, or a date that is out of range, the system shall return a list of all the date validation errors it failed 
  - Previously it would only return one date validation error with the priority being format>valid date>range
  - But now all three validations will be checked (and not short circuited) 
- If the user does not input a beginDate or endDate, the system shall only return this error message:
  - "**[insert property]** should not be null or undefined" 
  - It will not return the format, valid date, or range error messages, as there is no value provided to validate

### CODE REVIEW

File: src/dto/hourly-apportioned-emissions.params.dto.ts [hunk 1]
```diff
@@ -8,18 +12,41 @@ import { UnitType } from '../enums/unit-type.enum';
...
+import { IsInDateRange } from '../pipes/is-in-date-range.pipe';
+import { IsDateGreaterThanEqualTo } from '../pipes/is-date-greater.pipe';
 import { IsControlTechnology } from 'src/pipes/is-control-technology.pipe';
 
 export class HourlyApportionedEmissionsParamsDTO extends PaginationDTO {
-  @IsOptional()
-  @ApiPropertyOptional()
-  @Validate(DateValidation)
+  @ApiProperty()
+  @IsInDateRange('1995-01-01', {
```
Comment (jwhitehead77) [inline src/dto/hourly-apportioned-emissions.params.dto.ts]: consider making this a true date range check where you can pass a minDate and maxDate. both minDate and maxDate can be optional and if so then the validator only checks to see if >= to minDate or <= maxDate and if both are provided then checks both conditions.

File: src/dto/hourly-apportioned-emissions.params.dto.ts [hunk 2]
```diff
@@ -8,18 +12,41 @@ import { UnitType } from '../enums/unit-type.enum';
...
+  @IsDefined()
   beginDate: Date;
 
-  @IsOptional()
-  @ApiPropertyOptional()
-  @Validate(DateValidation)
+  @ApiProperty()
+  @IsDateGreaterThanEqualTo('beginDate', {
+    message:
+      'Please enter an end date that is equal to or greater than the begin date',
```
Comment (jwhitehead77) [inline src/dto/hourly-apportioned-emissions.params.dto.ts]: to keep message format regarding property names consistent change message to...
**'Please enter an $property that is greater than or equal to the beginDate'**
this uses the $property token for the property the validator is for and then beingDate becasue that is the related property that is being passed in as an argument

File: src/dto/hourly-apportioned-emissions.params.dto.ts [hunk 3]
```diff
@@ -8,18 +12,41 @@ import { UnitType } from '../enums/unit-type.enum';
...
+  })
+  @IsInDateRange('1995-01-01', {
```
Comment (jwhitehead77) [inline src/dto/hourly-apportioned-emissions.params.dto.ts]: consider making this a true date range check where you can pass a minDate and maxDate. both minDate and maxDate can be optional and if so then the validator only checks to see if >= to minDate or <= maxDate and if both are provided then checks both conditions.

**Review by jwhitehead77 — CHANGES_REQUESTED** [review]
see comments for changes

**Comment by junie-stigdon** [comment]
- updated date-range decorator to accept a min and max date
- updated error messages per requirements
- added swagger plugin

**Review by jwhitehead77 — APPROVED** [review]

**Pull request merged**